adapters/jwplayer/utils.go

```diff
@@ -0,0 +1,199 @@
+	PlacementID int `json:"placement_id,omitempty"`
+}
+
+// copied from appnexus.go appnexusImpExt
+type appnexusImpExt struct {
+	Appnexus appnexusImpExtParams `json:"appnexus"`
+}
+
+func GetAppnexusExt(placementId string) json.RawMessage {
+	id, conversionError := strconv.Atoi(placementId)
+	if conversionError != nil {
+		return nil
+	}
+
+	appnexusExt := &appnexusImpExt{
+		Appnexus: appnexusImpExtParams{
+			PlacementID: id,
+		},
+	}
+
+	jsonExt, jsonError := json.Marshal(appnexusExt)
+	if jsonError != nil {
+		return nil
+	}
+
+	return jsonExt
+}
+
+func ParsePublisherParams(publisher openrtb2.Publisher) *jwplayerPublisher {
+	var pubExt publisherExt
+	if err := json.Unmarshal(publisher.Ext, &pubExt); err != nil {
+		return nil
+	}
+
+	return &pubExt.JWPlayer
+}
+
+func ParseContentMetadata(content openrtb2.Content) ContentMetadata {
+	metadata := ContentMetadata{
+		Url:   content.URL,
+		Title: content.Title,
+	}
+
+	contentExt := ContentExt{}
+	if error := json.Unmarshal(content.Ext, &contentExt); error == nil {
+		metadata.Description = contentExt.Description
+	}
+
+	return metadata
+}
+
+func GetExistingJwpsegs(data []openrtb2.Data) []string {
+	for _, datum := range data {
+		if HasJwpsegs(datum) {
+			return ParseJwpsegs(datum.Segment)
+		}
+	}
+
+	return nil
+}
+
+func HasJwpsegs(datum openrtb2.Data) bool {
+	dataExt := DataExt{}
+	if error := json.Unmarshal(datum.Ext, &dataExt); error != nil {
+		return false
+	}
+
+	return datum.Name == jwplayerDomain && dataExt.Segtax == jwplayerSegtax && len(datum.Segment) > 0
+}
+
+func IsValidMediaUrl(rawUrl string) bool {
+	parsedUrl, error := url.ParseRequestURI(rawUrl)
+	if error != nil {
+		return false
+	}
+
+	isLocalFile := parsedUrl.Scheme == "file"
+	isLocalHost := parsedUrl.Opaque != ""
+	path := parsedUrl.Path
+	isRelativePath := parsedUrl.Scheme == "" && len(path) > 2 && path[0:1] == "/" && path[1:2] != "/"
+
+	return !isLocalFile && !isLocalHost && !isRelativePath
+}
+
+type EndpointTemplateParams struct {
+	SiteId      string
+	MediaUrl    string
+	Title       string
+	Description string
+}
+
+func BuildTargetingEndpoint(endpointTemplate *template.Template, siteId string, contentMetadata ContentMetadata) string {
+	if endpointTemplate == nil {
+		return ""
+	}
+
+	mediaUrl := url.QueryEscape(contentMetadata.Url)
+	title := url.QueryEscape(contentMetadata.Title)
+	description := url.QueryEscape(contentMetadata.Description)
+
+	endpointParams := EndpointTemplateParams{
+		SiteId:      siteId,
+		MediaUrl:    mediaUrl,
+		Title:       title,
+		Description: description,
+	}
+
+	reqUrl, macroResolveErr := macros.ResolveMacros(endpointTemplate, endpointParams)
+	if macroResolveErr != nil {
+		return ""
+	}
+
+	return reqUrl
+}
+
+func ParseJwpsegs(segments []openrtb2.Segment) []string {
+	jwpsegs := make([]string, len(segments))
+	for index, segment := range segments {
+		jwpsegs[index] = segment.Value
+	}
+
+	return jwpsegs
+}
+
+func GetAllJwpsegs(targeting TargetingData) []string {
```

Comment:
I think we are putting too much business logic in utils.  When I think of utils I think of generic functions that do transformations.